tests/imhttp-post-payload-basic-auth.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+# This is part of the rsyslog testbench, licensed under ASL 2.0
+
+. ${srcdir:=.}/diag.sh init
+generate_conf
+IMHTTP_PORT="$(get_free_port)"
+add_conf '
+template(name="outfmt" type="list") {
+  property(name="msg")
+  constant(value=" - ")
+  constant(value="{ baz: ")
+  property(name="$!metadata!httpheaders!baz")
+  constant(value=", daddle: ")
+  property(name="$!metadata!httpheaders!daddle")
+  constant(value=", fiddle: ")
+  property(name="$!metadata!httpheaders!fiddle")
+  constant(value=" }")
+  constant(value=" - ")
+  constant(value=" !metadata: ")
+  property(name="$!metadata")
+  constant(value="\n")
+}
+
+#
+# note htpasswd file contains entry:
+# user1:1234
+#
+module(load="../contrib/imhttp/.libs/imhttp"
+       ports="'$IMHTTP_PORT'")
+input(type="imhttp" endpoint="/postrequest"
+      ruleset="ruleset"
+      addmetadata="on"
+      basicauthfile="testsuites/htpasswd")
```

Comment:
`make distcheck` environment is differently, you will not find the files inside the subdir of the current working directory (because the build is moved out of if). you need to use the `$srcdir` variable to get to the tarball extract point. I think this will work (but may be wrong):
```
basicauthfile="'$srcdir'/testsuites/htpasswd"
```

Note that you can run `make distcheck` also locally, what will get you the same issue. If you place a bash call inside the test, you see the directory structure first hand.

Reply:
thanks @rgerhards, making the changes!

Reply:
think that was it! one of the tests failed, but looks unrelated to the change.

Reply:
Looks like flakes - we sometimes have timing issues. I restart and guess that will solve the issue.